### PR TITLE
W-11086334: Migrate lightweight usages of ApplicationDescriptor creation in runtime deployment

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

**Comment by jcgrethe** [comment]
--validate

File: modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/artifact/MuleDeployableProjectModelBuilder.java [hunk 1]
```diff
@@ -6,480 +6,20 @@
...
-
-  public static final String CLASS_LOADER_MODEL_VERSION_120 = "1.2.0";
-  public static final String CLASS_LOADER_MODEL_VERSION_110 = "1.1.0";
-
-  private static final String GROUP_ID = "groupId";
-  private static final String ARTIFACT_ID = "artifactId";
-
-  private final File projectFolder;
+@Deprecated
+public class MuleDeployableProjectModelBuilder
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/artifact/MuleDeployableProjectModelBuilder.java:19]: this is internal, can't this just be removed?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Comment by jcgrethe** [comment]
--validate

**Review by elrodro83 — APPROVED** [review]

**Comment by jcgrethe** [comment]
--validate

File: modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/plugin/PluginClassLoaderConfigurationAssembler.java [hunk 2]
```diff
@@ -64,7 +65,16 @@ sharedProjectDependencies, attributeToList(bundleDependency.getPackages()),
     this.bundleDependencies = bundleDependencies;
     this.bundleDependency = bundleDependency;
     this.ownerDescriptor = ownerDescriptor;
-    this.pluginPatchesResolver = pluginPatchesResolver;
+    Collection<PluginPatchesResolver> resolverRegistered =
+        new SpiServiceRegistry().lookupProviders(PluginPatchesResolver.class, this.getClass().getClassLoader());
+    if (resolverRegistered.size() > 1) {
+      throw new MuleRuntimeException(createStaticMessage("There is more than 1 PluginPatchesResolver implementation registered"));
```
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/plugin/PluginClassLoaderConfigurationAssembler.java:71]: add the found items to the exception message

File: modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/plugin/MuleArtifactPatchingModel.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
 import java.util.Map;
 import java.util.Optional;
 
-class MuleArtifactPatchingModel {
+public class MuleArtifactPatchingModel {
```
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/plugin/MuleArtifactPatchingModel.java:27]: where is this used? can it be moved along to retain its current visibility?

File: modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultLightweightApplicationFactoryTestCase.java [hunk 4]
```diff
@@ -6,84 +6,126 @@
...
 import io.qameta.allure.Feature;
 import io.qameta.allure.Issue;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
 
 @Feature(APP_CREATION)
-@Issue("W-11911617")
-public class ApplicationsCreatedWithDeprecatedDescriptorClassTestCase extends AbstractMuleTestCase {
+@Issue("W-11086334")
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultLightweightApplicationFactoryTestCase.java:63]: add the lightweight stroy

File: modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/domain/DefaultLightweightDomainFactoryTestCase.java [hunk 5]
```diff
@@ -21,27 +25,40 @@
...
 import io.qameta.allure.Issue;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
 
 @Feature(DOMAIN_CREATION)
-@Issue("W-11911617")
-public class DomainsCreatedWithDeprecatedDescriptorClassTestCase extends AbstractMuleTestCase {
+@Issue("W-11086334")
+public class DefaultLightweightDomainFactoryTestCase extends AbstractMuleTestCase {
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/domain/DefaultLightweightDomainFactoryTestCase.java:54]: add lightweight story

File: modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/plugin/DeployableDescriptorsWithPatchedPluginsTestCase.java [hunk 6]
```diff
@@ -35,7 +45,7 @@
 
 @Story(PATCHED_ARTIFACT_DESCRIPTORS)
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/plugin/DeployableDescriptorsWithPatchedPluginsTestCase.java:46]: add feature

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Comment by jcgrethe** [comment]
--validate

**Pull request merged**